File: crates/calm-core/tests/fixtures/multi_lang_workspace/ruby/handlers.rb
# typed: false

class AlphaHandler
  def process
    "alpha"
  end
end

class BetaHandler
  def process
    "beta"
  end
end

class Dispatcher
  def run(use_alpha)
    handler = use_alpha ? AlphaHandler.new : BetaHandler.new
    route(handler)
  end

  def route(handler)
    case handler
    when AlphaHandler
      handler.process
    when BetaHandler
      handler.process
    else
      "unknown"
    end
  end
end

File: crates/calm-core/tests/fixtures/multi_lang_workspace/kotlin/src/main/kotlin/com/example/Handlers.kt
package com.example

class AlphaHandler {
    fun process(): String {
        return "alpha"
    }
}

class BetaHandler {
    fun process(): String {
        return "beta"
    }
}

class Dispatcher {
    fun run(useAlpha: Boolean): String {
        val handler: Any = if (useAlpha) AlphaHandler() else BetaHandler()
        return dispatch(handler)
    }

    private fun dispatch(handler: Any): String {
        return when (handler) {
            is AlphaHandler -> handler.process()
            is BetaHandler -> handler.process()
            else -> "unknown"
        }
    }
}
